dashboard/app/views/levels/_dialog.html.haml

```diff
@@ -3,18 +3,23 @@
 
 - unless app == 'external'
   .buttons{class: @level.properties['submittable'] == 'true' ? 'submittable' : nil}
-    - if local_assigns[:previous_button]
+    - if local_assigns[:previous_page_button]
       %a.btn.btn-large.btn-primary.previousPageButton
         =t('previous_page')
-    - if local_assigns[:next_button]
+    - if local_assigns[:next_level_link]
+      %a{href: local_assigns[:next_link_level], class: 'btn btn-large btn-primary next-lesson nextLevelButton', style: !!local_assigns[:show_next_page_link] ? nil : 'display:none'}
```

Comment:
maybe a comment here about this button being dynamically unhidden in the JS? reading through this haml alone, i was confused why we were adding a conditionally hidden element inside an `if` rather than just adding the `show_next_page_link` check to the `if`. took me a minute to remember you showed that interaction with clicking the dialog button converting "submit" into "next level".

Reply:
yeah -- I was having trouble figuring out a good way of showing users a link to the next page and this is the best I could come up with. Will definitely add a comment!